cleverhans/model_abs.py

```diff
@@ -0,0 +1,73 @@
+from abc import ABCMeta
+from keras.models import Model
+
+
+class ModelAbstraction:
+    """
+    An abstract interface for a model that exposes the required functionalities
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, model):
+        """
+        :param model: A function that takes a symbolic input and returns the
+                      symbolic output for the model's predictions.
+        """
+        raise NotImplementedError('Model abstraction not implementated.')
+
+    def get_hidden(self, layer):
```

Comment:
No. I'm suggesting that we make it possible to build the model's fprop graph on top of more than one symbolic tensor. For adversarial training, we will usually want to have one fprop graph built on a symbolic tensor representing the clean training examples, and another graph built on a symbolic tensor representing the adversarial examples. Does that make sense?